KenticoListApp/TodoList.Api.Tests/ListItemsControllerTest.cs

```diff
@@ -0,0 +1,218 @@
+
+            Assert.That(actualListItem, Is.EqualTo(expected).Using(_comparer));
+        }
+
+        [Test]
+        public void Get_ById_IsOfCorrectStatusCode()
+        {
+            IHttpActionResult actionResult = _controller.GetAsync(Guid.Empty).Result;
+            HttpResponseMessage responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+        }
+
+        [Test]
+        public void Get_ReturnsTestItems()
+        {
+            var expectedListItems = new List<ListItem>
+            {
+                new ListItem(new Guid("00000000-0000-0000-0000-000000000000"), "text"),
+                new ListItem(new Guid("10000000-0000-0000-0000-000000000000"), "giraffe"),
+                new ListItem(new Guid("20000000-0000-0000-0000-000000000000"), "updated"),
+            };
+
+            IHttpActionResult actionResult = _controller.GetAsync().Result;
+            HttpResponseMessage responseMessage= actionResult.ExecuteAsync(CancellationToken.None).Result;
+            List<ListItem> actualListItems;
+            responseMessage.TryGetContentValue(out actualListItems);
+
+            Assert.That(actualListItems, Is.EqualTo(expectedListItems).Using(_comparer));
+        }
+
+        [TestCase("")]
+        [TestCase("    ")]
+        [TestCase(null)]
+        public void Post_ItemWithNullText_ReturnsErrorMessage(string postedText)
+        {
+            var listItem = new ListItem(Guid.Empty, postedText);
+
+            IHttpActionResult actionResult = _controller.Post(listItem).Result;
+            HttpResponseMessage responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            HttpError error;
+            responseMessage.TryGetContentValue(out error);
+            string actualMessage = error.Message;
+
+            Assert.That(actualMessage, Is.EqualTo("Text is null or empty"));
+        }
+
+        [Test]
+        public void Post_WithNullArguments_ReturnsErrorMessage()
+        {
+            IHttpActionResult actionResult = _controller.Post(null).Result;
+            HttpResponseMessage repsonseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            HttpError error;
+            repsonseMessage.TryGetContentValue(out error);
+            string actualMessage = error.Message;
+
+            Assert.That(actualMessage, Is.EqualTo("Item is null"));
+        }
+
+        [Test]
+        public void Post_WithNullArguments_IsOfCorrectStatusCode()
+        {
+            IHttpActionResult actionResult = _controller.Post(null).Result;
+            HttpResponseMessage responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+        }
+
+        [Test]
+        public void Post_ItemWithNullText_IsOfCorrectStatusCode()

```

Comment:
You can merge this with `Post_ItemWithNullText_ReturnsErrorMessage` as they are almost same.